static/src/javascripts-legacy/projects/common/modules/onward/tech-feedback.js

```diff
@@ -110,23 +51,89 @@ define([
         }
     }
 
-    /**
-     * the link in the footer adds some of the values to the hash so feedback can use it later.  Those values
-     * override those at the time the email is sent.
-     */
+    function toggleFormVisibility() {
+        $.forEachElement("#feedback-category>option", function(elem){
+            if(elem.selected && elem.value !== "nothing"){
+                document.getElementById(elem.value).classList.add("feedback__form--selected");
+            } else if(elem.value !== "nothing") {
+                document.getElementById(elem.value).classList.remove("feedback__form--selected");
+            }
+        });
+    }
+
+    function mandatoryCheck(elem) {
```

Comment:
This function does too much, and appears to be called for different reasons by different parts of the code, and hence it is not clear from the name what it is supposed to be doing, and what it returns. Can we split it up?